src/components/screens/new-questionnaire-screen.tsx

```diff
@@ -7,48 +22,90 @@ import { KeyboardReturn } from "@mui/icons-material";
 import UserRoleUtils from "src/utils/user-role-utils";
 import type { Question, QuestionOption } from "src/types/index";
 import strings from "src/localization/strings";
+import { useLambdasApi } from "src/hooks/use-api";
 
 /**
  * New Questionnaire Screen component
- * Includes a card (new-questionnaire-card.tsx) to create questions for a new questionnaire
+ * Includes a card (new-questionnaire-card.tsx) to create options for a new questionnaire
  */
 const NewQuestionnaireScreen = () => {
   const adminMode = UserRoleUtils.adminMode();
-  
+  const { questionnaireApi } = useLambdasApi();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
   const [questionnaireTitle, setQuestionnaireTitle] = useState("");
+  const [questionnaireDescription, setQuestionnaireDescription] = useState("");
+  const [options, setOptions] = useState<Question[]>([]);
+  const [passScoreValue, setPassScoreValue] = useState(0);
 
   const handleQuestionTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
     setQuestionnaireTitle(event.target.value);
   };
 
-  /**
-   * State and functions to handle the questions and options
-   */
-  const [questions, setQuestions] = useState<Question[]>([]);
+  const handleDescriptionChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuestionnaireDescription(event.target.value);
+  };
+
+  const handlePassScoreSliderChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setPassScoreValue(Number(event.target.value));
+  };
 
   /**
    * Functions to add new question to Questionnaire
-   * @param questionText 
-   * @param options 
+   * @param question
+   * @param options
    */
-  const handleAddQuestionSubmit = (
-    questionText: string,
-    options: QuestionOption[]
-  ) => {
-    setQuestions((prevQuestions: Question[]) => [...prevQuestions, { questionText, options }]);
+  const handleAddQuestionSubmit = (question: string, options: QuestionOption[]) => {
+    setOptions((prevQuestions: Question[]) => [...prevQuestions, { question, options }]);
   };
 
   const handleDeleteQuestion = (index: number) => {
```

Comment:
Missing Docs

Reply:
Is this deleting questions or options, we need consistant naming

Reply:
Well this is to delete already built question before Questionnaire is saved. In UI admin needs to build questions one by one. By saving question it is then rendered in preview. So this is a possibility to delete question from preview before the hole Questionnaire is saved.

Also I now think our naming specs suck... Options in options...

Reply:
From what I understand we are removing the option from state here, but not actually deleting anything that has been persisted to the DB. Might be worth trying to clarify that in the naming if possible. Maybe its not necessary- your call